### PR TITLE
Added support for setting Catalog viewers via API

### PR DESCRIPTION
**Pull request by clintonb**
[ECOM-4086](https://openedx.atlassian.net/browse/ECOM-4086)

### CODE REVIEW

**Comment by clintonb** [comment]
@bderusha @peter-fogg please review

File: course_discovery/apps/catalogs/models.py [hunk 1]
```diff
@@ -52,6 +56,52 @@ def contains(self, course_ids):  # pylint: disable=unused-argument
...
+            value (Iterable): Collection of `User` objects.
+
+        Raises:
+            TypeError: The given value is not iterable, or is a string.
+
+        Returns:
+            None
+        """
+        if isinstance(value, str) or not isinstance(value, Iterable):
+            raise TypeError('Viewers must be a non-string iterable containing User objects.')
```
Comment (peter-fogg) [inline course_discovery/apps/catalogs/models.py:88]: We haven't discussed this at all yet, but we might want to do some investigation of [mypy](http://mypy-lang.org/) to check this statically.

**Comment by peter-fogg** [comment]
:+1:

File: course_discovery/apps/catalogs/models.py (hunk 1 above)
Comment (clintonb) [inline course_discovery/apps/catalogs/models.py:88]: Interesting. Were we under less of a deadline, I would explore `mypy` further. For now, this is safe as-is, especially given the fact that the serializer will do a lot of the validation for us.

**Pull request merged**